src/site/content/en/fast/codelab-adapt-video-to-image-serving-based-on-network-quality/index.md

```diff
@@ -0,0 +1,210 @@
+---
+layout: codelab
+title: Adapt video to image serving based on network quality
+authors:
+  - mihajlija
+description: |
+  Learn how to use the Network Information API to adapt
+  your content based on the network quality.
+glitch: adaptive-serving-netinfo-starter
+related_post: adaptive-serving-based-on-network-quality
+---
+
+In this codelab, you’ll learn how to adapt your content based on the network
+quality. This page's background video will load only when users are on a fast
+network. On slower networks, an image will load instead.
+
+The
+[Network Information API](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation)
+enables you to access information about the user's connection quality. You will
+use its `effectiveType` property to decide when to serve a video and when to
+serve an image. `effectiveType` can be `'slow-2g'`, `'2g'`, `'3g'`, or `'4g'`.
+
+## Step 1: Check connection type
+
+The background video is currently specified on line 22 of `index.html` and then
+lazy-loaded in `script.js`.
+
+To load it conditionally, first check if connection type detection is supported.
+In `script.js` add an if statement that tests whether the `navigator.connection`
+object exists and whether it has the `effectiveType` property.
+
+Then add an if statement to check the `effectiveType` of the network. To load
+the video only on the fastest connections, it’s enough to check the connection
+for `4g` and in any other case load the image.
+
+```js
+if (navigator.connection && !!navigator.connection.effectiveType) {
+  if (navigator.connection.effectiveType === '4g') {
+    // video loading code
+  } else {
+    // image loading code
+  }
+}
+```
+
+Wrap the existing video loading code in an else statement, so that video will
+still load in browsers that don’t support the Network Information API.
+
+```js
+if (navigator.connection && !!navigator.connection.effectiveType) {
+  if (navigator.connection.effectiveType === '4g') {
+    // video loading code
+  } else {
+    // image loading code
+  }
+} else {
+  const video = document.getElementById('coverVideo');
+  const videoSource = video.getAttribute('data-src');
+  video.setAttribute('src', videoSource);
+
+  video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+}
+```
+
+## Step 2: Load video
+
+If the `effectiveType` is `'4g'`, you can use the video loading code from the
+beginning of the codelab.
+
+```js
+if (navigator.connection.effectiveType === '4g') {
+  const video = document.getElementById('coverVideo');
+  const videoSource = video.getAttribute('data-src');
+  video.setAttribute('src', videoSource);
+  video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+} else {
+  // image loading code
+}
+```
+
+Here’s how the video loading code works: the video source is specified inside
```

Comment:
```suggestion
Here’s how the video loading code works: the video source is specified using the
```

Reply:
Alternately, maybe explain a little more here. Like:

.. code works: the `<video>` tag doesn't download or display anything at first because its `src` attribute is not set. The video URL to load is specified using the `data-src` attribute.